### DISCUSSION TITLE
[Help] Enter Key closes modal

### DISCUSSION THREAD

**Discussion by veebzz** [fomantic/Fomantic-UI#1660] [Behaviors (Api, Form Validation, Visibility)] [answered]
I am trying to create a modal with a form inside it

# Help Wanted
Functionality I want to make:
1. Register anchor opens modal
2. Modal should only close on clicking the background dimmer, x icon, or cancel button, then clear the form.
3. The submit button performs a custom function on click(not shown in jsfiddle), and hides modal after done.

According to Modal docs if a false is passed back to onApprove, then the modal should not hide. If I open the modal, and press the enter key, the modal closes.
I have tried adding the keyboardShortcuts: false to the modal, form, buttons, and it did not seem to work. Im not sure what else to do to prevent the enter key from closing the modal, or restrict it so that the enter button imitates clicking submit button.

## Testcase
https://jsfiddle.net/phcoyk2w/36/

**Comment by lubber-de** [comment, marked as answer]
The Browsers impement their own logic when pressing the enter key inside a form and trigger the submit by themselves if there is at least one submit button. By default each `<button>` is a submit button.
Your `keyboardShortcuts: false` approach was quite right, however to also prevent the browsers default logic, use `div` buttons instead
See https://jsfiddle.net/lubber/hajkt2rb/13/

We had a similar question at #1632

**Comment by veebzz** [comment]
That makes alot of sense! Thanks @lubber-de !
